sdk/src/types/block/output/rent.rs

```diff
@@ -140,3 +148,44 @@ impl<T: Rent, const N: usize> Rent for [T; N] {
         self.iter().map(|elem| elem.weighted_bytes(config)).sum()
     }
 }
+
+pub struct MinimumStorageDeposit {
+    config: RentStructure,
+    token_supply: u64,
+    builder: BasicOutputBuilder,
+}
+
+impl MinimumStorageDeposit {
+    pub fn new(config: RentStructure, token_supply: u64) -> Self {
+        Self {
+            config,
+            token_supply,
+            builder: BasicOutputBuilder::new_with_amount(Output::AMOUNT_MIN).add_unlock_condition(
+                AddressUnlockCondition::new(Address::from(Ed25519Address::from([0; Ed25519Address::LENGTH]))),
+            ),
+        }
+    }
+
+    pub fn with_native_tokens(mut self, native_tokens: impl Into<Option<NativeTokens>>) -> Self {
+        if let Some(native_tokens) = native_tokens.into() {
+            self.builder = self.builder.with_native_tokens(native_tokens);
+        }
+        self
+    }
+
+    pub fn with_storage_deposit_return(mut self, return_address: Address) -> Result<Self, Error> {
+        self.builder = self
+            .builder
+            .add_unlock_condition(StorageDepositReturnUnlockCondition::new(
+                return_address,
+                Output::AMOUNT_MIN,
+                self.token_supply,
+            )?)
+            .add_unlock_condition(ExpirationUnlockCondition::new(return_address, 1)?);
```

Comment:
Should we add it automatically or have another with_ method ?

Reply:
Should be separated, it's not always wanted

Reply:
Done